### PR TITLE
Use kubectl describe foo name

### PR DESCRIPTION
**Pull request by fejta**
If we do not include the `-o name` then the command output includes a ` (dry run)` suffix.

/assign @smukherj1 @nlopezgi 

This seems to be causing issues in recent kubectl versions (https://github.com/bazelbuild/rules_k8s/pull/193#issuecomment-431026365).

* Formatting fixes while I'm in these files
* Run `delete || true` before running create, in case zombie resources exist from a prior run.
* Logging the steps before they run to assist debugging

### CODE REVIEW

**Comment by erain** [comment]
I thought `kubectl get foo name` is more commonly used than `kubectl get foo/name`?  Example: [kubectl Cheat Sheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/)

**Comment by fejta** [comment]
Maybe `kubectl get foo name` and `kubectl get foo/name` are both fine, but it doesn't like `kubectl get foo foo/name`

**Comment by fejta** [comment]
Looks like the problem is in the actual bazel rules... Let me see if I can figure this out...

**Comment by fejta** [comment]
These e2e tests are the WORST :sweat_smile:

**Comment by fejta** [comment]
Okay I think I have it!

Looks like the problem here is that `kubectl create --dry-run -f "%{unresolved}"` now outputs (dry run):
`deployment.apps/hello-grpc-staging created (dry run)`
But we want the name without the (dry run) suffix:
`deployment.apps/hello-grpc-staging created`

We can fix this by adding an `-o name` flag.

**Comment by fejta** [comment]
PASS! :fireworks: :champagne: :confetti_ball:

File: examples/hellohttp/e2e-test.sh [hunk 1]
```diff
@@ -40,33 +41,35 @@ check_msg() {
...
-    else
-	echo "Bad substitution should fail!"
-	exit 1
-    fi
+  echo Checking a bad substitution...
+  if ! bazel run examples/hellohttp:error-on-run; then
+    echo "Success, substitution failed."
+    return 0
+  else
+    echo "Bad substitution should fail!"
```
Comment (chrislovecnm) [inline examples/hellohttp/e2e-test.sh]: If I am a grammar critic it could read a tad better, but a nit pick

File: examples/hellohttp/e2e-test.sh [hunk 2]
```diff
@@ -40,33 +41,35 @@ check_msg() {
 }
 
 edit() {
-   ./examples/hellohttp/${LANGUAGE}/edit.sh "$1"
+  echo Setting $LANGUAGE to "$1"
```
Comment (chrislovecnm) [inline examples/hellohttp/e2e-test.sh]: Please quote language var. We might want to lint with shellcheck

File: examples/hellogrpc/e2e-test.sh [hunk 3]
```diff
@@ -24,44 +24,49 @@ if [[ -z "${1:-}" ]]; then
 fi
 
 get_lb_ip() {
-    kubectl --namespace=${USER} get service hello-grpc-staging \
-	-o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+  kubectl --namespace=${USER} get service hello-grpc-staging \
```
Comment (chrislovecnm) [inline examples/hellogrpc/e2e-test.sh]: Please put quotes around all variables

File: examples/todocontroller/e2e-test.sh [hunk 4]
```diff
@@ -18,66 +18,72 @@ set -e
 LANGUAGE="$1"
 
 function get_lb_ip() {
-    kubectl --namespace=${USER} get service hello-grpc-staging \
-	-o jsonpath='{.status.loadBalancer.ingress[0].ip}'
+  kubectl --namespace=${USER} get service hello-grpc-staging \
+    -o jsonpath='{.status.loadBalancer.ingress[0].ip}'
```
Comment (chrislovecnm) [inline examples/todocontroller/e2e-test.sh]: Again missing quotes around a bash var

**Comment by chrislovecnm** [comment]
Well that did not work 😂😂

**Comment by chrislovecnm** [comment]
I can do a follow-up PR as well. @fjeta thoughts?

**Comment by chrislovecnm** [comment]
@fejta I removed my blocking review, so up to you if you want to fix my recommendations or not.  I can do a follow-up PR as I mentioned.

**Comment by chrislovecnm** [comment]
This is blocking testing big time. @erain / @fejta / @nlopezgi can someone merge?

**Pull request merged**